ProjectEureka/project-eureka-flutter
# PE-114 | Check for existing chat

What I Did:
- Updated code for joining the chat from the More Details page. Now it will check if a chat with the recipient on this question already exists.
- Updated code on the chat screen. Now when joining the chat, it will pass the group chat ID, so that we don't need to parse through chats to find one on the chat_screen.dart file
- Cleaned code, separated joining the chat from more details page into functions

How to Test:
- Go through different questions and start a chat with different users through the chat icon or Answer question button, and click on Chat icons in different answers of the same user. It must bring to the same existing chat or create a new one.
- Test if messages and images are being sent properly
- Try leaving the chat screen by clicking on Question Details or companion's name
- Test if video communication works fine
- Check if Flutter shows any errors in the logs
